src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs

```diff
@@ -565,7 +565,8 @@ private void OnWorkspaceStatusChanged(object sender, EventArgs args)
                 }
 
                 // ask editor to refresh light-bulb when workspace solution status is changed
-                this.SuggestedActionsChanged?.Invoke(this, EventArgs.Empty);
+                // 🐉 Currently this event is disabled since it can cause unexpected light bulb dismissal
+                //this.SuggestedActionsChanged?.Invoke(this, EventArgs.Empty);
```

Comment:
not a fan of this.

Reply:
I can't find any case where firing this event makes a difference.

Reply:
so my understanding is that it's for the use case of someone putting their caret on something.  They have a particular set of lightbulb options then (or have none whatsoever).  but they then do something like add a reference, and now we expect the lightbulb to change.

If that works already because things like focus is lost/added, then this may also be for the something changed on disk scenario.

*that said*, i'm totally ok relaxing this in 17.3 and then seeing if there are any problems we discover using teh feature ourselves :)  my guess is that hte scenarios are niche, and having a more reliable lightbulb trumps them.

Reply:
note: if we're not going to fire this, i would recommend excising a lot more of the complexity in this type related to eventing and state management :)